### PR TITLE
fix(take): finish the iterable as soon as requested items are took

### PR DESCRIPTION
**Pull request by GerkinDev**
`asyncTake` waits for the next iteration item before completing, which may never be yield. The break condition have been slightly modified & moved right after the yield to allow early exits.

While this issue should not happen for sync iterables, those modifications have been reflected on the sync version too.

### CODE REVIEW

**Comment by conartist6** [comment]
Looks good, but you created a bug with the `n === 0` corner case. Now we need  a bail-out check for `n===0` before the for loop. Also once you add that line you'll  need to test it for coverage purposes, though that should be as simple as adding a second line to that first test.

**Comment by GerkinDev** [comment]
All set ! I used `<` & `>=` comparison operators to mitigate cases where non-int `n` would cause the iterable to never complete.

**Comment by conartist6** [comment]
Oh, sorry, but I'd prefer to leave the comparisons as `===`. I have a separate system for validating argument types.

**Comment by GerkinDev** [comment]
Oh yeah, I see this now, `$iterableCurry`. So, fixed. Nice tooling pipeline btw.

File: src/impls/$take/__tests__/$take.test.ts [hunk 1]
```diff
@@ -10,4 +10,30 @@ describe($`take`, () => {
...
+    }),
+  );
+  it(
+    'completes immediately after taking the first n values',
+    $async(() => {
+      $async;
+      function* twoItemsThenNever() {
+        yield 1;
+        yield 2;
+        // Never yields 3.
```
Comment (conartist6) [inline src/impls/$take/__tests__/$take.test.ts]: One last change. Could we just throw an error here instead of causing the test runner to time out?

Comment (GerkinDev) [reply src/impls/$take/__tests__/$take.test.ts]: Yes, we could, but that case should never happens anyway as long as the implementation is correct. But you're the boss, on it

**Comment by conartist6** [comment]
Thanks! I'm working on rebuilding the tooling around the `$` stuff right now actually, as some important parts of it (like watch mode) are broken. I do very much like that it allows contributors to edit once and end up with equally readable sync and async code though.

**Comment by GerkinDev** [comment]
I know you don't asked me for my opinion but I think the philosophy is good.

The 2 small problems I see though is that
* the `async.macro.cjs` file does not really exports the symbols used in tests, and it gives annoying red underline in vscode
* and maybe the contributing documentation could be improved a bit. It took me a couple of minutes to understand where I should put my modifications and how the tooling interacts with the code.

**Comment by conartist6** [comment]
I can fix both those things!

**Comment by conartist6** [comment]
The fix is released in `v7.2.1`. I appreciate your help and patience!
